src/plugins/discover/public/application/main/services/discover_app_state_container.ts

```diff
@@ -350,13 +350,29 @@ export function isEqualFilters(
  * Helper function to compare 2 different state, is needed since comparing filters
  * works differently
  */
-export function isEqualState(stateA: DiscoverAppState, stateB: DiscoverAppState) {
+export function isEqualState(
+  stateA: DiscoverAppState,
+  stateB: DiscoverAppState,
+  exclude: string[] = []
+) {
   if (!stateA && !stateB) {
     return true;
   } else if (!stateA || !stateB) {
     return false;
   }
+
   const { filters: stateAFilters = [], ...stateAPartial } = stateA;
   const { filters: stateBFilters = [], ...stateBPartial } = stateB;
-  return isEqual(stateAPartial, stateBPartial) && isEqualFilters(stateAFilters, stateBFilters);
+
+  const filteredStateAPartial = excludeKeys(stateAPartial, exclude);
+  const filteredStateBPartial = excludeKeys(stateBPartial, exclude);
+
+  return (
+    isEqual(filteredStateAPartial, filteredStateBPartial) &&
+    isEqualFilters(stateAFilters, stateBFilters)
+  );
```

Comment:
Nit: rather than declaring more vars and defining our own function for this, it might be cleaner to instead use lodash's `omit` inline:
```ts
  const { filters: stateAFilters = [], ...stateAPartial } = omit(stateA, exclude);
  const { filters: stateBFilters = [], ...stateBPartial } = omit(stateB, exclude);

  return isEqual(stateAPartial, stateBPartial) && isEqualFilters(stateAFilters, stateBFilters);
```

Reply:
That's right! thx, changed that and added tests for this function